mm2src/coins/eth/eip1559_gas_fee.rs

```diff
@@ -0,0 +1,469 @@
+struct BlocknativeBaseFee {
+    confidence: u32,
+    #[serde(rename = "baseFee")]
+    base_fee: BigDecimal,
+}
+
+struct BlocknativeEstimatedBaseFees {}
+
+impl BlocknativeEstimatedBaseFees {
+    /// Parse blocknative's base_fees in pending blocks : '[ "pending+1" : {}, "pending+2" : {}, ..]' removing 'pending+n'
+    fn parse_pending<'de, D>(deserializer: D) -> Result<Vec<Vec<BlocknativeBaseFee>>, D::Error>
+    where
+        D: Deserializer<'de>,
+    {
+        struct PendingBlockFeeParser;
+        impl<'de> Visitor<'de> for PendingBlockFeeParser {
+            type Value = Vec<Vec<BlocknativeBaseFee>>;
+
+            fn expecting(&self, formatter: &mut fmt::Formatter) -> fmt::Result {
+                formatter.write_str("[u32, BigDecimal]")
+            }
+
+            fn visit_seq<A: SeqAccess<'de>>(self, mut seq: A) -> Result<Self::Value, A::Error> {
+                let mut pending_block_fees = Vec::<Vec<BlocknativeBaseFee>>::new();
+                while let Some(fees) = seq.next_element::<HashMap<String, Vec<BlocknativeBaseFee>>>()? {
+                    if let Some(fees) = fees.iter().next() {
+                        pending_block_fees.push(fees.1.clone());
+                    }
+                }
+                Ok(pending_block_fees)
+            }
+        }
+        deserializer.deserialize_any(PendingBlockFeeParser {})
+    }
+}
+
+#[allow(dead_code)]
+#[derive(Debug, Deserialize)]
+struct BlocknativeBlockPricesResponse {
+    system: String,
+    network: String,
+    unit: String,
+    #[serde(rename = "maxPrice")]
+    max_price: u32,
+    #[serde(rename = "currentBlockNumber")]
+    current_block_number: u32,
+    #[serde(rename = "msSinceLastBlock")]
+    ms_since_last_block: u32,
+    #[serde(rename = "blockPrices")]
+    block_prices: Vec<BlocknativeBlockPrices>,
+    #[serde(
+        rename = "estimatedBaseFees",
+        deserialize_with = "BlocknativeEstimatedBaseFees::parse_pending"
+    )]
+    estimated_base_fees: Vec<Vec<BlocknativeBaseFee>>,
+}
+
+lazy_static! {
+    static ref BLOCKNATIVE_GAS_API_AUTH_TEST: String =
+        std::env::var("BLOCKNATIVE_GAS_API_AUTH_TEST").unwrap_or_default();
+}
+
+#[allow(dead_code)]
+pub(super) struct BlocknativeGasApiCaller {}
+
+#[allow(dead_code)]
+impl BlocknativeGasApiCaller {
+    const BLOCKNATIVE_GAS_API_URL: &'static str = "https://api.blocknative.com/gasprices";
+    const BLOCKNATIVE_GAS_PRICES_CALL: &'static str = "blockprices";
+    const BLOCKNATIVE_GAS_PRICES_PARAMS: &'static str =
+        "confidenceLevels=10&confidenceLevels=50&confidenceLevels=90&withBaseFees=true";
+
+    fn get_blocknative_gas_api_url() -> (String, Vec<(&'static str, &'static str)>) {
+        let url = format!(
+            "{}/{}?{}",
+            Self::BLOCKNATIVE_GAS_API_URL,
+            Self::BLOCKNATIVE_GAS_PRICES_CALL,
+            Self::BLOCKNATIVE_GAS_PRICES_PARAMS
+        );
+        let headers = vec![("Authorization", BLOCKNATIVE_GAS_API_AUTH_TEST.as_str())];
```

Comment:
I added a config value for the gas api in MM2.json. However I'd like to leave the api key const like it is for now. This is a temp solution, for testing gas api, until we switch to use komodo proxy instead of direct connection to the api. With the komodo proxy the authentication would be different and this const would be removed